### PR TITLE
Sso configuration

### PR DESCRIPTION
**Pull request by shivani170**

### CODE REVIEW

**Review by rashmirai21 — APPROVED** [review]

File: src/components/login/Login.tsx [hunk 1]
```diff
@@ -1,34 +1,153 @@
...
+    }
+
+    renderSSOLoginPage() {
+        let search = this.props.location.search;
+
+        return <div className="login__control">
+                <img src={dt} alt="login" className="login__dt-logo" width="170px" height="120px" />
+                <p className="login__text">Your tool for Rapid, Reliable & Repeatable deployments</p>
+                {this.state.loginList.map((item) => {
+                    return <a href={`/orchestrator/auth/login?return_url=${this.state.continueUrl}`} className="login__google flex">
```
Comment (nishant-d) [inline src/components/login/Login.tsx:102]: base url is hard-code

File: src/components/login/SSOLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+        }
+    }
+    saveNewSSO(): void {
+        this.setState({ saveLoading: true });
+
+        let configJSON: any = {};
+        try {
+            configJSON = yamlJsParser.parse(this.state.ssoConfig.config.config);
+        } catch (error) {
+            //Invalid YAML, couldn't be parsed to JSON. Show error toast
```
Comment (nishant-d) [inline src/components/login/SSOLogin.tsx:126]: error handling missing

File: src/components/login/SSOLogin.tsx [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+    }
+
+    onLoginConfigSave(): void {
+        this.setState({ saveLoading: true });
+
+        let configJSON: any = {};
+        try {
+            configJSON = yamlJsParser.parse(this.state.ssoConfig.config.config);
+        } catch (error) {
+            //Invalid YAML, couldn't be parsed to JSON. Show error toast
```
Comment (nishant-d) [inline src/components/login/SSOLogin.tsx:168]: show error to the user

File: src/components/login/SSOLogin.tsx [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+                    config: {
+                        name: this.state.ssoConfig.config.name,
+                        type: this.state.ssoConfig.config.type,
+                        id: this.state.ssoConfig.config.id,
+                        config: value,
+                    },
+                }
+            });
+        } catch (error) {
+        }
```
Comment (nishant-d) [inline src/components/login/SSOLogin.tsx:258]: error handling

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: src/components/login/SSOLogin.tsx (hunk 3 above)
Comment (rashmirai21) [reply src/components/login/SSOLogin.tsx:170]: Done

File: src/components/login/SSOLogin.tsx (hunk 2 above)
Comment (rashmirai21) [reply src/components/login/SSOLogin.tsx:126]: Done

File: src/components/login/Login.tsx (hunk 1 above)
Comment (rashmirai21) [reply src/components/login/Login.tsx]: Done

File: src/components/login/SSOLogin.tsx (hunk 4 above)
Comment (rashmirai21) [reply src/components/login/SSOLogin.tsx]: try-catch not required here as there is no conversion of yaml to json. Therefore, removed it

**Review by nishant-d — APPROVED** [review]

**Pull request merged**